ftw/maintenanceserver/server.py

```diff
@@ -39,10 +39,20 @@ def __init__(self, request, client_address, server):
     do_POST = SimpleHTTPRequestHandler.do_GET
 
     def do_OPTIONS(self):
-        self.send_response(200)
+        self.send_response(200, method='OPTIONS')
         self.send_header('Allow', 'GET,HEAD,POST,OPTIONS')
+        self.send_header('Cache-Control', 'no-cache')
         self.end_headers()
 
+    def send_response(self, code, message=None, method=None):
+        # Always send "503 Service Unavailable" instead of "200 OK" so that
+        # maintenance servers do not cache the response.
```

Comment:
The maintenance server isn't the one doing the caching :wink:
-> `caching proxies do not cache maintenance server responses`.